RallyHereIntegration/Source/RallyHereEditor/Private/RallyHereEditorLootWidget.cpp

```diff
@@ -0,0 +1,1123 @@
+	COPY_OPTIONAL_VALUE(Quantity, Loot.Get()->, LootUpdateRequest.)
+	COPY_OPTIONAL_VALUE(Active, Loot.Get()->, LootUpdateRequest.)
+	COPY_OPTIONAL_VALUE(SortOrder, Loot.Get()->, LootUpdateRequest.)
+	COPY_OPTIONAL_VALUE(DropWeight, Loot.Get()->, LootUpdateRequest.)
+	COPY_OPTIONAL_VALUE(FillInNewOrder, Loot.Get()->, LootUpdateRequest.)
+	COPY_OPTIONAL_VALUE(AllowPartialBundles, Loot.Get()->, LootUpdateRequest.)
+	COPY_OPTIONAL_VALUE(RequiredItemId, Loot.Get()->, LootUpdateRequest.)
+	COPY_OPTIONAL_VALUE(RequiredItemCount, Loot.Get()->, LootUpdateRequest.)
+	COPY_OPTIONAL_VALUE(StackLimit, Loot.Get()->, LootUpdateRequest.)
+	COPY_OPTIONAL_VALUE(QuantityType, Loot.Get()->, LootUpdateRequest.)
+	COPY_OPTIONAL_VALUE(QuantityMultiInventoryItemId, Loot.Get()->, LootUpdateRequest.)
+	COPY_OPTIONAL_VALUE(IsClaimableByClient, Loot.Get()->, LootUpdateRequest.)
+	COPY_OPTIONAL_VALUE(TimeFrameId, Loot.Get()->, LootUpdateRequest.)
+	COPY_OPTIONAL_VALUE(UseInventoryBucket, Loot.Get()->, LootUpdateRequest.)
+	COPY_OPTIONAL_VALUE(XpQuantityTransformType, Loot.Get()->, LootUpdateRequest.)
+	COPY_OPTIONAL_VALUE(CurrentPricePointId, Loot.Get()->, LootUpdateRequest.)
+	COPY_OPTIONAL_VALUE(PreSalePricePointId, Loot.Get()->, LootUpdateRequest.)
+	COPY_OPTIONAL_VALUE(HardQuantityMaximum, Loot.Get()->, LootUpdateRequest.)
+	LootUpdateRequest.SetInventorySelectorType(Loot.Get()->GetInventorySelectorType());
+	LootUpdateRequest.SetInventoryOperation(Loot.Get()->GetInventoryOperation());
+	LootUpdateRequest.SetVendorId(Loot.Get()->GetVendorId());
+}
+
+FReply SRallyHereEditorLootRow::OnRevertLoot()
+{
+	HydrateLootUpdateRequest();
+	ItemIdText.Get()->SetText(FText::FromString(FString::Printf(TEXT("(%s)"), *Loot.Get()->GetLootId().ToString(EGuidFormats::DigitsWithHyphensLower))));
+	RequiredItemIdText.Get()->SetText(FText::FromString(FString::Printf(TEXT("(%s)"), *Loot.Get()->GetRequiredItemId(FGuid()).ToString(EGuidFormats::DigitsWithHyphensLower))));
+	QuantityMultiInventoryItemIdText.Get()->SetText(FText::FromString(FString::Printf(TEXT("(%s)"), *Loot.Get()->GetQuantityMultiInventoryItemId(FGuid()).ToString(EGuidFormats::DigitsWithHyphensLower))));
+	NameText.Get()->SetText(FText::FromString(FString::Printf(TEXT("%s"), *Loot.Get()->GetName(""))));
+	DescriptionText.Get()->SetText(FText::FromString(FString::Printf(TEXT("%s"), *Loot.Get()->GetDescription(""))));
+	LootDirtyBox.Get()->Invalidate(EInvalidateWidget::Visibility);
+	CustomData.Get()->SetCustomData(Loot.Get()->GetCustomData());
+	return FReply::Handled();
+}
+
+void SRallyHereEditorLootRow::OnCustomDataUpdated()
+{
+	LootUpdateRequest.SetCustomData(CustomData.Get()->GetCustomData());
+	IsDirty = true;
+	LootDirtyBox.Get()->Invalidate(EInvalidateWidget::Visibility);
+}
+
+FReply SRallyHereEditorLootRow::OnUpdateLoot()
+{
+	auto Request = TSandboxUpdateLoot::Request();
+
+	Request.AuthContext = FModuleManager::Get().LoadModuleChecked<FRallyHereEditorModule>(FRallyHereEditorModule::GetModuleName()).GetAuthContext();
```

Comment:
The Auth Context get + logged in check feels like something that could be wrappered better?  Its in a lot of functions.